Player Update

cdgonz115 opened the pull request:
Created the User Interface and updated the player prefab as well as created the win scene

cdgonz115 committed:
Updated Player Shot

Updated the player shooting mechanic to more closely follow the documented mechanic wanted, look at documentation for full details.

cdgonz115 committed:
Adams Feedback

Made it so that the player can sort of edge the shot and then continue the charge if the shot wasn’t charged 100%, and hasn't decayed all the way planning to put some sort of ui element to let the player know how the charged up it is, also changed the times of the frames to 120 so its more noticeable.

cdgonz115 committed:
Basic Functionality

The basic switching between panels is working as well some of the other methods such as quit, play, load scene, etc. Also added the current boss scenes to the build settings so i could load them.

cdgonz115 committed:
Updates

Almost finished with all of the UI functuonality, just have to add the remaping, add sounds so that the sound functionality can work and fix the pausing issue.

cdgonz115 committed:
Merge branch 'master' into Fixed_Shooting

cdgonz115 committed:
Update

Got rid of an unecessary varaible so that UI elements to signal the shot charging is easier

cdgonz115 committed:
Fixes

Got some fixes done based on daniels and adamas feedback reguarding the resetting of the charge

cdgonz115 committed:
Progress

Added the remaping functionality however there is a big bug i mist fix

cdgonz115 committed:
Updated UI

UI now has functional remaping for keyboard as well as the majority of the sprites needed for it

cdgonz115 committed:
Merge Fixed Shooting

Merged Fixed shootign branch

cdgonz115 committed:
Sprite Update

Updated the sprite for the bullet the player shoots

cdgonz115 committed:
Merge branch 'Fixed_Shooting' into EndScenes

cdgonz115 committed:
Revert "Merge branch 'Fixed_Shooting' into EndScenes"

This reverts commit 835f8b6e8d95b5c5cbd0d6ab6c62810eb9bbd045, reversing
changes made to 01c37d7595c5d534ae106edd3ed3aaaeb6aa6efe.

cdgonz115 committed:
Merge branch 'User_Interface' into EndScenes

cdgonz115 committed:
End Scenes Added

Added the win scene as well as the win and loss UI lements and functionality

fire-lemon committed:
Added requisite packages  for 2D animations

cdgonz115 committed:
Merge branch 'EndScenes' into User_Interface

cdgonz115 committed:
Player Update

Finalized the UI and update the player prefab with Marcos's sprites and aniamtions ass well as changed some scripts to account for the update. Also created the win scene.

Doughboy02 commented on Assets/Scripts/Scenes/WinScene/BossesStare.cs:
```diff
@@ -0,0 +1,14 @@
+    public bool inverted;
+    void Update()
+    {
+        var VectorToPlayer = (Vector3)(PlayerHealth.singleton.transform.position - transform.position).normalized * (inverted? -1:1 );
```
Just picky thing here, don't use vars unless you absolutely have to

cdgonz115 committed:
Charge Bug Fix

Fixed a bug that allowed the player to charge the attack ability while swinging the sword, thanks connor for the catch
